### PR TITLE
Allow rendering MJML components to text and text-ish HTML

### PR DESCRIPTION
**Pull request by sjchmiela**
Newly introduced `MjmlRenderingContext` exposes "mode" of rendering — `html | text`
- if rendering to HTML, nothing changes — we still render `<mj-...>` elements which are later rendered by `mjml2html`
- if rendering to text, components can now "override" output and decide to render something else, eg.
	```
	<div>
		{children}: {rest.href}
	</div>
	```
    , as is the case with `MjmlButton`. Or `null` in case of `MjmlHead`. Or `<hr />` in case of `MjmlDivider`.

**Important note:** components aren't expected to render to string only. To simplify development we run `htmlToText` on the output later. This lets us both: benefit from formatting logic of `html-to-text` and keep the HTML escape hatches in email templates.

To render in different modes:
```tsx
const email = /* ... */;

// Render HTML

const htmlMjml = renderToMjml(
  <MjmlRenderingContext.Provider value={{ mode: "html" }}>
    {email}
  </MjmlRenderingContext.Provider>
);

const { html } = mjml2html(htmlMjml, {
  validationLevel: "soft",
  minify: false,
});

// Render text

// 1. Render React to MJML
const textMjml = renderToMjml(
  <MjmlRenderingContext.Provider value={{ mode: "text" }}>
    {email}
  </MjmlRenderingContext.Provider>
);

// 2. Parse MJML (as if it was HTML) to text
const text = htmlToText(textMjml, {
  wordwrap: false,
});
```

## Preview

With little changes to the template we get:

Zrzut ekranu 2022-10-21 o 16 41 54

---

Also added `pnpm dev`, `pnpm build`, `pnpm tsc` and removed `build.js` which seems to have been superseded by `tsc`

### CODE REVIEW

**Comment by vpontis** [comment]
@sjchmiela very cool, some ideas:

1. Can we update `renderToMjml` to take care of rendering the `Context` for us? → `renderToMjml(email, 'text' | 'html')`
2. `htmlToText` → `mjmlToHtml`

**Comment by sjchmiela** [comment]
1. Indeed, thanks for noticing that! I tried to do it this way too, but back then I tried to move `html2text` into `mjml-react`  too which we can't do, because it depends on Node environment and we need to keep `mjml-react` Node-independent.
2. Not sure what you're suggesting here. If you're suggesting to include `mjmlToHtml` step somewhere, it seems to me we can't do that in this approach — components rendered in `text` mode render invalid MJML, eg. `<div>` as direct child of `<mj-section>` and `mjmlToHtml` skips invalid MJML.
    
    An alternative approach to this pull request's proposal would have been overriding what _all_ components render and ensuring they only return strings in `text` mode (I _think_ this would be possible). Then we wouldn't need neither `mjmlToHtml` nor `html2text` — `ReactDOM.renderToStaticMarkup` would do the trick (all elements would render to text so the result would be text). I don't recommend going this way because while more explicit and clearer it would also be very limiting (no HTML in email templates) and requiring us to override much more code (explicitness comes with a cost).

    Or maybe it's a typo you suggested to add a utility in `server` that would be called `mjmlToText`? Or maybe to change modes from `html | text` to `mjml | html`?

**Comment by vpontis** [comment]
2. You have a function called `htmlToText` in your description, I suggest you call it `mjmlToText`.

**Comment by sjchmiela** [comment]
Got it. It's an exported function from `html-to-text` called `htmlToText`. `mjmlToHtml` doesn't make too much sense since the output is text (no HTML). And it seems funny to me to do
```tsx
import { htmlToText as mjmlToText } from "html-to-text";
```

I'd rather comment on each `htmlToText(mjmlString)` occurrence sth like "MJML is semi-valid HTML so we can use htmlToText to transform it to text".

**Comment by vpontis** [comment]
Ah your code snippet didn't include the import so I assumed it was a function you created. Thanks for explaining. 

Also whoops I meant to suggest `mjmlToText` as the new name.

File: src/mjml-button.tsx [hunk 1]
```diff
@@ -9,7 +10,16 @@ export const MjmlButton = ({
 }: React.PropsWithChildren<
   MjmlButtonProps & PaddingProps & ClassNameProps & HrefProps & BorderProps
 >) => {
-  return React.createElement("mj-button", handleMjmlProps(rest), children);
+  return renderTo({
+    html: React.createElement("mj-button", handleMjmlProps(rest), children),
+    text: children ? (
```
Comment (vpontis) [inline src/mjml-button.tsx]: Wdyt of calling this `mjml / simpleHtml` rather than `html / text`?

**Review by vpontis — APPROVED** [review]

File: src/mjml-button.tsx (hunk 1 above)
Comment (sjchmiela) [reply src/mjml-button.tsx]: I started to like textish HTML, reminds me of [bestish mensch](https://youtu.be/_BihD5SzOvg) haha

`mjml / simpleHtml` sound good to me, though `simpleHtml` lacks a bit of a pointer to what this is going to be used for, that is converting to text (which eg. explain why we wouldn't use any paddings, colors and everything is a div or a line break haha)

What do you think of `mjml / textHtml`? Or `htmlForText`? Doesn't make too big of a difference at the end of the day…

Replacing to `mjml / simpleHtml`, let me know if any of other suggestions sounds better to you!

Comment (vpontis) [reply src/mjml-button.tsx]: Yea your call!

**Pull request merged**